test/unit-tests/controller.test.ts

```diff
@@ -2701,29 +2699,30 @@ contract(
             )
           })
 
-          it('should revert withdraw collateral from a vault with an expired short otoken', async () => {
-            const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
-            assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
-
-            const collateralToWithdraw = createTokenAmount(10, usdcDecimals)
-            const actionArgs = [
-              {
-                actionType: ActionType.WithdrawCollateral,
-                owner: accountOwner1,
-                secondAddress: accountOwner1,
-                asset: usdc.address,
-                vaultId: vaultCounter.toNumber(),
-                amount: collateralToWithdraw,
-                index: '0',
-                data: ZERO_ADDR,
-              },
-            ]
-
-            await expectRevert(
-              controllerProxy.operate(actionArgs, {from: accountOwner1}),
-              'Controller: can not withdraw collateral from a vault with an expired short otoken',
-            )
-          })
+          // TODO: remove or change this
+          // it('should revert withdraw collateral from a vault with an expired short otoken', async () => {
+          //   const vaultCounter = new BigNumber(await controllerProxy.getAccountVaultCounter(accountOwner1))
+          //   assert.isAbove(vaultCounter.toNumber(), 0, 'Account owner have no vault')
+
+          //   const collateralToWithdraw = createTokenAmount(10, usdcDecimals)
+          //   const actionArgs = [
+          //     {
+          //       actionType: ActionType.WithdrawCollateral,
+          //       owner: accountOwner1,
+          //       secondAddress: accountOwner1,
+          //       asset: usdc.address,
+          //       vaultId: vaultCounter.toNumber(),
+          //       amount: collateralToWithdraw,
+          //       index: '0',
+          //       data: ZERO_ADDR,
+          //     },
+          //   ]
+
+          //   await expectRevert(
+          //     controllerProxy.operate(actionArgs, {from: accountOwner1}),
+          //     'Controller: can not withdraw collateral from a vault with an expired short otoken',
+          //   )
+          // })
```

Comment:
Can you test that if you withdraw collateral after expiry, you cannot withdraw more than the amount that settleVault would give you? (For a type 0 vault)

Reply:
Added tests for withdrawing more collateral than payout in an expired position, withdrawing some of the collateral and settling should still only give a sum of amounts equal to the payout

https://github.com/opynfinance/GammaProtocol/blob/6647517cf4bd30de0b3bf763c70e620a450e223b/test/unit-tests/controller.test.ts#L3640